### PR TITLE
drivers: sensors: Add driver for AHT20 temperature humidity sensor.

### PR DESCRIPTION
**Pull request by RafaelLeeImg**
Support AHT20 Temperature and Humidity Sensor with I2C bus

Webpage for the sensor is here:
[https://asairsensors.com/product/aht20-integrated-temperature-and-humidity-sensor/](https://asairsensors.com/product/aht20-integrated-temperature-and-humidity-sensor/)

Data sheet v1.0.03 is here:
[https://asairsensors.com/wp-content/uploads/2021/09/Data-Sheet-AHT20-Humidity-and-Temperature-Sensor-ASAIR-V1.0.03.pdf](https://asairsensors.com/wp-content/uploads/2021/09/Data-Sheet-AHT20-Humidity-and-Temperature-Sensor-ASAIR-V1.0.03.pdf)

Tested with physical sensor.
arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1 20210621 (release)

### CODE REVIEW

File: drivers/sensor/aht20/aht20.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	struct aht20_data *drv_data = dev->data;
+	float temperature;
+
+	switch (chan) {
+	case SENSOR_CHAN_HUMIDITY:
+		val->val1 = 0;
+		val->val2 = (drv_data->humidity) * 1000000LL / (1 << 20);
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		temperature = (float)drv_data->temperature / (1 << 20) * 200 - 50.0;
```
Comment (gmarull) [inline drivers/sensor/aht20/aht20.c:32]: please define magic numbers here

File: drivers/sensor/aht20/aht20.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+{
+	struct aht20_data *drv_data = dev->data;
+
+	drv_data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
+	if (drv_data->i2c == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!", DT_INST_BUS_LABEL(0));
+		return -EINVAL;
+	}
+
+	k_busy_wait(40000); // wait for 40ms
```
Comment (gmarull) [inline drivers/sensor/aht20/aht20.c:52]: reason? this will also delay system startup

File: drivers/sensor/aht20/aht20.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+	int ret = 0;
+	aht20_status sensor_status = { 0 };
+
+	{
```
Comment (gmarull) [inline drivers/sensor/aht20/aht20.c:57]: ?

**Review by gmarull — CHANGES_REQUESTED** [review]
Did an initial review, there are quite a few issues that need to be addressed.

File: drivers/sensor/aht20/aht20.c (hunk 2 above)
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c:49]: The sensor needs 40ms to initialize before any operation. Is there any good way to this during or after startup?

Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Replaced with k_sleep

File: drivers/sensor/aht20/Kconfig [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2016 Intel Corporation
```
Comment (teburd) [inline drivers/sensor/aht20/Kconfig:1]: Update to your own name and email here (or company) and year

File: drivers/sensor/aht20/aht20.c [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	struct aht20_data *drv_data = dev->data;
+
+	k_sleep(K_MSEC(40)); // wait for 40ms
+
+	int ret = 0;
+	aht20_status sensor_status = { 0 };
+
+	static uint8_t const inquire_status_seq[] = { AHT20_CMD_GET_STATUS };
+
+	ret = i2c_write_dt(&drv_data->bus, inquire_status_seq, sizeof(inquire_status_seq));
```
Comment (teburd) [inline drivers/sensor/aht20/aht20.c:56]: i2c_write_read_dt doesn't work for this sequence of doing a GET_STATUS and obtaining the status value?

File: drivers/sensor/aht20/aht20.c [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	}
+
+	ret = i2c_read_dt(&drv_data->bus, &sensor_status.all, AHT20_STATUS_LENGTH);
+	if (!sensor_status.cal_enable) {
+		static uint8_t const initialize_seq[] = { AHT20_CMD_INITIALIZE };
+		ret = i2c_write_dt(&drv_data->bus, initialize_seq, sizeof(initialize_seq));
+		if (0 != ret) {
+			LOG_ERR("Failed to inquire status");
+		}
+		k_busy_wait(10000); // wait for 10ms
```
Comment (teburd) [inline drivers/sensor/aht20/aht20.c:68]: 10ms is a long time, k_sleep is nicer here

File: drivers/sensor/aht20/aht20.c [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+	}
+
+	aht20_status sensor_status = { 0 };
+
+	// tested with AHT20, 40ms is enough for measuring, datasheet said wait 80ms
+	k_sleep(K_MSEC(40));
+	bool wait = false;
+
+	do {
+		if (wait) {
```
Comment (teburd) [inline drivers/sensor/aht20/aht20.c:98]: could remove wait altogether and check if the status.busy bit is true here I believe

File: drivers/sensor/aht20/aht20.h [hunk 8]
```diff
@@ -0,0 +1,48 @@
+/*
```
Comment (teburd) [inline drivers/sensor/aht20/aht20.h:1]: I'm not sure this header is really needed at all? All of this could be privately placed in the .c if you like.

**Review by teburd — CHANGES_REQUESTED** [review]
Primarily the k_busy_wait concerns me

File: drivers/sensor/aht20/aht20.c (hunk 6 above)
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in commit 1519d40bc59faf7be0b3d52484a21b3eb5d8a784.

File: drivers/sensor/aht20/aht20.c (hunk 7 above)
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in commit 1519d40bc59faf7be0b3d52484a21b3eb5d8a784.

**Review by teburd — CHANGES_REQUESTED** [review]
This looks pretty good, the comments need to use /* */ rather than // though, and I believe checkpatch will fail on those.

If you plan on contributing a lot to zephyr I recommend setting up checkpatch as a git hook.
See https://docs.zephyrproject.org/latest/contribute/index.html#coding-style

**Comment by RafaelLeeImg** [comment]
I used both clang-format and uncrustify to format my source code, but the space before the colon is introduced by uncrustify, if there is no proper tool to format code. I'm using version Uncrustify-0.72.0_f, is my version too old? Otherwise, it's better to fix the uncrustify config instead of modify the code format by hand.
```bash
uncrustify --replace --no-backup -l C -c $ZEPHYR_BASE/.uncrustify.cfg aht20.c
```

**Comment by ssekar15** [comment]
@RafaelLeeImg we are interested on this driver, do you have plan to continue with the PR !!

**Comment by RafaelLeeImg** [comment]
> @RafaelLeeImg we are interested on this driver, do you have plan to continue with the PR !!

Hi.
I would like to continue with the current PR. I've already formatted the code according to the coding style proposed by
```bash
uncrustify --replace --no-backup -l C -c $ZEPHYR_BASE/.uncrustify.cfg
```
But, at the stage, the CI still complains coding formats error.
I think make every tiny format modification and let CI tell me where I'm wrong is very inefficient.
Have the .uncrustify.cfg or CI script updated?

**Comment by RafaelLeeImg** [comment]
> @RafaelLeeImg we are interested on this driver, do you have plan to continue with the PR !!

**Comment by ssekar15** [comment]
@RafaelLeeImg Thanks for updates. Could you squash the multiple draft commits to single clean commit and rebase to latest please !!

**Comment by RafaelLeeImg** [comment]
I have opened an issue with the topic of coding style.
[https://github.com/zephyrproject-rtos/zephyr/issues/51973](https://github.com/zephyrproject-rtos/zephyr/issues/51973)

File: drivers/sensor/aht20/aht20.c [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		temperature = (float)drv_data->temperature / (1 << AHT20_FULL_RANGE_BITS) *
+				      AHT20_TEMPERATURE_RANGE -
+			      AHT20_TEMPERATURE_OFFSET;
+		val->val1 = (int32_t)temperature;
+		val->val2 = (temperature - val->val1) * 1000000LL;
+		break;
+	default:
+		break;
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in 3aea61cd2a4

File: drivers/sensor/aht20/aht20.c [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+static int aht20_init(const struct device *dev)
+{
+	struct aht20_data *drv_data = dev->data;
+
+	k_sleep(K_MSEC(40)); /* wait for 40ms */
+
+	int ret = 0;
+	aht20_status sensor_status = {0};
+
+	static uint8_t const inquire_status_seq[] = {AHT20_CMD_GET_STATUS};
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in 3aea61cd2a4

File: drivers/sensor/aht20/aht20.c [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+	ret = i2c_write_read_dt(&drv_data->bus, inquire_status_seq, sizeof(inquire_status_seq),
+				&sensor_status.all, AHT20_STATUS_LENGTH);
+	if (ret != 0) {
+		LOG_ERR("Failed to inquire status");
+	}
+
+	if (!sensor_status.cal_enable) {
+		static uint8_t const initialize_seq[] = {AHT20_CMD_INITIALIZE};
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in 3aea61cd2a4

File: drivers/sensor/aht20/aht20.c [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+static int aht20_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct aht20_data *drv_data = dev->data;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_AMBIENT_TEMP ||
+			chan == SENSOR_CHAN_HUMIDITY);
+
+	uint8_t tx_buf[] = {AHT20_CMD_TRIGGER_MEASURE, AHT20_TRIGGER_MEASURE_BYTE_0,
+			    AHT20_TRIGGER_MEASURE_BYTE_1};
+	uint8_t rx_buf[7] = {0};
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c:116]: Fixed in 3aea61cd2a4
and add
```C
LOG_INF("AHT20: Inquire status successfully");
```

File: drivers/sensor/aht20/aht20.c [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+		return -EIO;
+	}
+
+	aht20_status sensor_status = {0};
+
+	/* tested with AHT20, 40ms is enough for measuring, datasheet said wait 80ms */
+	k_sleep(K_MSEC(40));
+
+	while (1) {
+		rc = i2c_read_dt(&drv_data->bus, rx_buf, AHT20_STATUS_LENGTH + AHT20_READ_LENGTH);
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in 3aea61cd2a4

File: drivers/sensor/aht20/aht20.c [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+	/* crc8: 8 bits */
+	drv_data->humidity = sys_get_be24(rx_buf + 1);
+	drv_data->humidity >>= 24 - AHT20_FULL_RANGE_BITS;
+	drv_data->temperature = sys_get_be24(rx_buf + 3);
+	drv_data->temperature &= (1L << AHT20_FULL_RANGE_BITS) - 1;
+
+	uint8_t crc = crc8(rx_buf, AHT20_STATUS_LENGTH + AHT20_READ_LENGTH - 1, AHT20_CRC_POLY,
+			   AHT20_CRC_INIT, false);
+
+	if (crc != rx_buf[AHT20_STATUS_LENGTH + AHT20_READ_LENGTH - 1]) {
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in 3aea61cd2a4

File: drivers/sensor/aht20/aht20.c (hunk 12 above)
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in cfa26ce2a2f52b0ddd5f4cfaeb447606aff2b601
```C
uint8_t rx_buf[AHT20_TOTAL_READ_LENGTH] = {0};
```

File: drivers/sensor/aht20/aht20.c [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+		break;
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		temperature = (float)drv_data->temperature / (1 << AHT20_FULL_RANGE_BITS) *
+				      AHT20_TEMPERATURE_RANGE -
+			      AHT20_TEMPERATURE_OFFSET;
+		val->val1 = (int32_t)temperature;
+		val->val2 = (temperature - val->val1) * 1000000LL;
+		break;
+	default:
+		-ENOTSUP;
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Fixed in cfa26ce2a2f52b0ddd5f4cfaeb447606aff2b601
```C
	default:
		return -ENOTSUP;
```

File: drivers/sensor/aht20/aht20.c [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_AMBIENT_TEMP ||
+			chan == SENSOR_CHAN_HUMIDITY);
+
+	uint8_t tx_buf[] = {AHT20_CMD_TRIGGER_MEASURE, AHT20_TRIGGER_MEASURE_BYTE_0,
+			    AHT20_TRIGGER_MEASURE_BYTE_1};
+	uint8_t rx_buf[7] = {0};
+
+	int rc = i2c_write_dt(&drv_data->bus, tx_buf, sizeof(tx_buf));
+
```
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c:115]: Fixed in cfa26ce2a2f52b0ddd5f4cfaeb447606aff2b601
Is the blank line convention a personal preference or a project guide line?

This is not described in
[https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style](https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style)
not in zephyr/.clang-format
neither in zephyr/scripts/checkpatch.pl

File: drivers/sensor/aht20/aht20.c [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+			LOG_INF("Initialization successful");
+		}
+		k_sleep(K_MSEC(10));
+	}
+	return ret;
+}
+
+static int aht20_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_AMBIENT_TEMP ||
```
Comment (ssekar15) [inline drivers/sensor/aht20/aht20.c:107]: please move logical code part after variable declaration

Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c]: Both ixed in 82c64bf0dab4b88ef507d814bc70b795e95262fe.

File: drivers/sensor/aht20/Kconfig (hunk 4 above)
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/Kconfig]: Fixed in Fixed in 82c64bf0dab4b88ef507d814bc70b795e95262fe.

**Review by ssekar15 — DISMISSED** [review]
Looks good to me.

File: drivers/sensor/CMakeLists.txt [hunk 18]
```diff
@@ -111,6 +111,7 @@ add_subdirectory_ifdef(CONFIG_ADC_CMP_NPCX	nuvoton_adc_cmp_npcx)
 add_subdirectory_ifdef(CONFIG_TACH_IT8XXX2	ite_tach_it8xxx2)
 add_subdirectory_ifdef(CONFIG_VCMP_IT8XXX2	ite_vcmp_it8xxx2)
 add_subdirectory_ifdef(CONFIG_PCNT_ESP32	pcnt_esp32)
+add_subdirectory_ifdef(CONFIG_AHT20		aht20)
```
Comment (MaureenHelm) [inline drivers/sensor/CMakeLists.txt:114]: Please move this up for alpha order

File: drivers/sensor/Kconfig [hunk 19]
```diff
@@ -261,4 +261,6 @@ source "drivers/sensor/ite_vcmp_it8xxx2/Kconfig"
 
 source "drivers/sensor/pcnt_esp32/Kconfig"
 
+source "drivers/sensor/aht20/Kconfig"
```
Comment (MaureenHelm) [inline drivers/sensor/Kconfig:264]: Please move this up for alpha order

File: drivers/sensor/aht20/Kconfig [hunk 20]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022 Rafael Lee
+# SPDX-License-Identifier: Apache-2.0
+
+config AHT20
+	bool "AHT20 Temperature and Humidity Sensor"
+	depends on I2C
```
Comment (MaureenHelm) [inline drivers/sensor/aht20/Kconfig:6]:
```suggestion
	default y
	depends on DT_HAS_ASAIR_AHT20_ENABLED
	select I2C
```

File: drivers/sensor/aht20/aht20.c [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+		break;
+	}
+	return 0;
+}
+
+static int aht20_init(const struct device *dev)
+{
+	struct aht20_data *drv_data = dev->data;
+
+	k_sleep(K_MSEC(40)); /* wait for 40ms */
```
[the remark this answers is not in the document]
Comment (MaureenHelm) [reply drivers/sensor/aht20/aht20.c:84]: Why is the sleep needed?

File: drivers/sensor/aht20/aht20.c [hunk 22]
```diff
@@ -0,0 +1,166 @@
...
+	}
+	if (!sensor_status.calibrated) {
+		LOG_INF("Calibrated bit is 0, need initialization");
+		ret = i2c_write_dt(&drv_data->bus, initialize_seq, sizeof(initialize_seq));
+		if (ret) {
+			LOG_ERR("Initialization filaed");
+		} else {
+			LOG_INF("Initialization successful");
+		}
+		k_sleep(K_MSEC(10));
```
Comment (MaureenHelm) [inline drivers/sensor/aht20/aht20.c:100]: How was the sleep duration chosen?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add a devicetree binding, add this sensor to `tests/drivers/build_all/sensor/`, and address the compliance check failures.

File: drivers/sensor/aht20/aht20.c (hunk 22 above)
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c:100]: Described in AHT20 datasheet v1.1

https://files.seeedstudio.com/wiki/Grove-AHT20_I2C_Industrial_Grade_Temperature_and_Humidity_Sensor/AHT20-datasheet-2020-4-16.pdf

5.4 Sensor reading process

1. Wait **40ms** after power-on. Before reading the temperature and humidity values, ﬁrst check whether the calibration enable bit Bit [3] of the status word is 1 (you can get a byte of status word by sending 0x71). If not 1, need to send 0xbe command (for initialization), this command parameter has two bytes, the ﬁrst byte is 0x08, the second byte is 0x00, and then wait for **10ms**.

File: drivers/sensor/Kconfig (hunk 19 above)
Comment (RafaelLeeImg) [reply drivers/sensor/Kconfig]: Done.

File: drivers/sensor/CMakeLists.txt (hunk 18 above)
Comment (RafaelLeeImg) [reply drivers/sensor/CMakeLists.txt]: Done.

File: drivers/sensor/aht20/aht20.c (hunk 21 above)
[the remark this answers is not in the document]
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c:84]: Described in AHT20 datasheet v1.1

https://files.seeedstudio.com/wiki/Grove-AHT20_I2C_Industrial_Grade_Temperature_and_Humidity_Sensor/AHT20-datasheet-2020-4-16.pdf

5.4 Sensor reading process

Wait **40ms** after power-on. Before reading the temperature and humidity values, ﬁrst check whether the calibration enable bit Bit [3] of the status word is 1 (you can get a byte of status word by sending 0x71). If not 1, need to send 0xbe command (for initialization), this command parameter has two bytes, the ﬁrst byte is 0x08, the second byte is 0x00, and then wait for **10ms**.

**Comment by RafaelLeeImg** [comment]
How to generate this symbol DT_HAS_ASAIR_AHT20_ENABLED ?
I didn't find any clue.

**Comment by ssekar15** [comment]
> DT_HAS_ASAIR_AHT20_ENABLED

^^ drop on kconfig.
please check example of dts binding in dts/bindings/sensor

add vendor prefix in dts/bindings/vendor-prefixes.txt
add dts entry for this sensor on tests/drivers/build_all/sensor/i2c.dtsi
 
 
 It seems humidity value is needs to be corrected according to datasheet it has to be multiplied by 100
 ```
		-val->val2 = (drv_data->humidity) * 1000000LL / (1 << AHT20_FULL_RANGE_BITS);
		+val->val2 = ((drv_data->humidity) * 1000000LL / (1 << AHT20_FULL_RANGE_BITS)) * 100;
```

**Comment by RafaelLeeImg** [comment]
I have wrote a test demo, the data format is right. Datasheet wrote RH[%]=(S_rh/2^20)*100%, 100% is 1.0.

**Comment by RafaelLeeImg** [comment]
H.val1 = 0, H.val2 = 335938, T.val1 = 14, T.val2 = 633667;
H = 0.335938, T=14.633667;
H = 33.59%, T = 14.63;
H = 33.5938%, T = 14.63;

**Comment by RafaelLeeImg** [comment]
Is there any further modifications required?

File: drivers/sensor/aht20/aht20.c [hunk 23]
```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2022 Rafael Lee <rafaellee.img@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT aosong_aht20
```
Comment (ssekar15) [inline drivers/sensor/aht20/aht20.c:7]: Please stick to the product details asair, aht20 according to datasheet.

Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c:7]: I prefer to think that ASAIR is a brand name.
I'm sure that aosong is the company's name.

What's your suggesstion, use the company's name or the brand name?

AHT20 is listed here
[http://www.aosong.com/productslist-1.html](http://www.aosong.com/productslist-1.html)
[http://www.aosong.com/products-61.html](http://www.aosong.com/products-61.html)


DHT humidity and temperature sensor with which with driver in Zephyr.
[http://www.aosong.com/productslist.aspx?id=1&page=2](http://www.aosong.com/productslist.aspx?id=1&page=2)
[http://www.aosong.com/products-104.html](http://www.aosong.com/products-104.html)

zephyr/drivers/sensor/dht/dht.c
```C
#define DT_DRV_COMPAT aosong_dht
```

Comment (MaureenHelm) [reply drivers/sensor/aht20/aht20.c:7]: It should be the company name. Is there a binding for this sensor in the Linux kernel?

File: drivers/sensor/aht20/aht20.c [hunk 24]
```diff
@@ -0,0 +1,166 @@
...
+{
+	uint8_t tx_buf[] = {AHT20_CMD_TRIGGER_MEASURE, AHT20_TRIGGER_MEASURE_BYTE_0,
+			    AHT20_TRIGGER_MEASURE_BYTE_1};
+	uint8_t rx_buf[AHT20_TOTAL_READ_LENGTH] = {0};
+	aht20_status sensor_status = {0};
+	struct aht20_data *drv_data = dev->data;
+	int rc = 0;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_AMBIENT_TEMP ||
+			chan == SENSOR_CHAN_HUMIDITY);
```
Comment (MaureenHelm) [inline drivers/sensor/aht20/aht20.c:115]: Return an errno

File: drivers/sensor/aht20/aht20.c [hunk 25]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+static const struct sensor_driver_api aht20_driver_api = {
+	.sample_fetch = aht20_sample_fetch,
+	.channel_get = aht20_channel_get,
+};
+
+#define AHT20_INIT(n)                                                                              \
+	static struct aht20_data aht20_data_##n = {                                                \
+		.bus = I2C_DT_SPEC_INST_GET(n),                                                    \
```
Comment (MaureenHelm) [inline drivers/sensor/aht20/aht20.c:161]: Move this to a const config struct

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
The compliance check failure still needs to be addressed

File: drivers/sensor/aht20/aht20.c (hunk 23 above)
Comment (RafaelLeeImg) [reply drivers/sensor/aht20/aht20.c:7]: > It should be the company name. Is there a binding for this sensor in the Linux kernel?

Aosong is the company name.
I don't know whether those sensor drivers exist in Linux kernel. I mean there is a driver from the same company in the Zephyr repository already.
[https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/dht/dht.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/sensor/dht/dht.c)

**Comment by ssekar15** [comment]
@RafaelLeeImg ping

**Pull request closed without merging**